### ISSUE TITLE
[FEATURE] Launch LEGUP JAR with Executable File (Windows)

### ISSUE THREAD

**Issue by TrevorBrunette** [Bram-Hub/Legup#123]
## Feature Description
Running a .jar file is okay for development, however in production it is generally seen as bad practice. Any program should come in a .exe executable file for Windows and an ELF file for UNIX-like systems. We should create an executable file which does not require the user to have java and java environment variables, i.e., it should come bundled with an appropriate JRE and use that. 

An executable file delivered this way will take up much more disk space and increase the time to downloaded for the user, however optimizing for file size and potentially using an installed JRE and downloading one if one does not exist would be an entirely separate feature. 

Primarily, development focus should be on creating an executable for Windows 10 (which should also work on Windows 11) as that is the operating system with the most market-share by far. Launch4j may assist in this endeavor greatly. Another feature to add in addition to this would be a GitHub action to automatically generate the program JAR file and  corresponding executable(s) whenever a push is made to the main or dev branches.

**Comment by charlestian23** [comment]
While development should primarily be focused on Windows, it would be best to also find a solution for macOS, as a good percentage of Professor Bram's students also use a Mac laptop.

**Comment by TrevorBrunette** [comment]
I've been looking into the issue more - it seems like the main way to accomplish this shipping on Windows is to create an installer which installs the JRE, executable, and any other dependencies like any other program. This is the "best" that a production version of this can be made without creating a custom EXE packaging tool for Legup. However, this is quite inconvenient for development considering how often there are updates. Reinstalling this on multiple systems for testing each version update could become quite cumbersome.

The current implementation that I have with launch4j allows for the EXE to be shipped (and then I suppose installed) with a JRE, and alternatively falls back on the Java program provided in the path (and somehow in the system registry as well it seems? I removed the Java parts from my path and it still worked but when I removed Java from my system it could not launch). When running the program, if there is no applicable Java executable, the program opens the [Java download page](https://java.com/en/download/) and prompts the user to either install Java or repair their installation of Java. This seems like it could suffice for now for Windows. All I would need to do is to make a script to automate the generation of this EXE upon build. 

I do not yet have any knowledge of programs which could be used to generate an installation EXE file which installs the software into "C:\Program Files", but I know they do exist. If anyone has any knowledge of software which is capable of this and is easily usable in scripting, please comment explaining and linking it.

We would need another person to work on this with me who uses a MacOS system to develop an executable for it. If this applies to anyone reading this and you believe you could get this working (and perhaps even help create a script to automate this upon releasees), please self-assign this issue and discuss this with me on this project Discord.

TLDR: What exactly do we want for production? An installer which would install Legup and its dependencies (more of a production release of the program) or an EXE which prompts the user to install Java if it cannot run? Including the Java runtime within the EXE seems like much more work than those options, but I can work on it if need be.

**Comment by TrevorBrunette** [comment]
I created an [Inno Setup](https://jrsoftware.org/isinfo.php) script to create an installer for Windows for this. Inno generates an executable which installs the program into `C:\Program Files (x86)\Legup\` alongside a usable JRE using an installer wizard. It allows for the display of the license and other information during install. It seems as if someone was also using Inno four years ago to try to do this but did not include any error checking for the user not having Java installed.

After install, the program cannot be ran by a user normally - it must be run as administrator. As described [here](https://github.com/Bram-Hub/Legup/issues/135), it will crash due to the generation or writing to of the logfile in a directory where the user does not have write permissions.
